### PR TITLE
[cloud_functions] Update example, README, depend on web plugin

### PR DESCRIPTION
**Pull request by sbeitzel**
## Description

This change updates the [README.md](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_functions/cloud_functions_web/README.md) with the correct version number of the `cloud_functions` plugin that imports the `cloud_functions_web` plugin. It also updates the `cloud_functions` example application with a web implementation.

## Related Issues 

flutter/flutter#45299

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by sbeitzel** [comment]
Ugh. Kotlin.

File: packages/cloud_functions/cloud_functions/pubspec.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: cloud_functions
 description: Flutter plugin for Cloud Functions.
-version: 0.4.1+8
+version: 0.4.1+9
 homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/cloud_functions/cloud_functions
 
 flutter:
```
Comment (hterkelsen) [inline packages/cloud_functions/cloud_functions/pubspec.yaml]: also add a `web:` section with a `default_package:` like here: https://github.com/flutter/plugins/blob/master/packages/url_launcher/url_launcher/pubspec.yaml#L16

Comment (sbeitzel) [reply packages/cloud_functions/cloud_functions/pubspec.yaml]: Ah, so that's how that happens. Okay, will do.

**Comment by sbeitzel** [comment]
The build checks probably won't pass until [PR 1910](https://github.com/FirebaseExtended/flutterfire/pull/1910) gets merged. And, you know, never deploy on Friday.

**Comment by sbeitzel** [comment]
Just looked at pub.dev and I realized that I'm going to have to bump the version number on `package:cloud_functions_web` one more time, to make the updated README get published. Since it's just a documentation change, I would think that 1.0.2+1 would be a good version, but previous analyzer checks have barfed on that version numbering style, insisting on a whole point increment (e.g. 1.0.3). Any thoughts?

**Comment by hterkelsen** [comment]
I think a "+1" release will be fine. If the analyzer complains we can merge and see if we can fix the pubspec analyzer, since "+1" releases should be allowed

**Comment by sbeitzel** [comment]
Yeah, here's the failure message:

`Running version check for changed packages
packages/cloud_functions/cloud_functions_web/pubspec.yaml incorrectly updated version.
HEAD: 1.0.2+1, master: 1.0.2.
Allowed versions: {2.0.0: NextVersionType.BREAKING_MAJOR, 1.1.0: NextVersionType.MINOR, 1.0.3: NextVersionType.PATCH}`

**Comment by hterkelsen** [comment]
It looks like the check only allows `+N` versions for versions that are `0.X.Y`. That's fine, we can just bump to the next patch version: `1.0.3`

**Pull request merged**